crypto/asn1/asn_mime.c

```diff
@@ -627,8 +627,12 @@ static int multi_split(BIO *bio, const char *bound, STACK_OF(BIO) **ret)
                 if (bpart == NULL)
                     return 0;
                 BIO_set_mem_eof_return(bpart, 0);
-            } else if (eol)
-                BIO_write(bpart, "\r\n", 2);
+            } else if (eol) {
+                if ((flags & CMS_BINARY) == 0 || (flags & SMIME_ASCIICRLF) != 0)
```

Comment:
Are any of these paths related to the SMIME_ASCIICRLF logic being hit?

Reply:
Good question!
It turns out that I confused SMIME_ASCIICRLF and SMIME_CRLFEOL.
Extended tests confirmed that; after corrections this should be fine now.